src/coreclr/src/jit/gentree.cpp

```diff
@@ -10566,16 +10736,30 @@ void Compiler::gtDispConst(GenTree* tree)
                 else if ((tree->AsIntCon()->gtIconVal > -1000) && (tree->AsIntCon()->gtIconVal < 1000))
                 {
                     printf(" %ld", dspIconVal);
-#ifdef TARGET_64BIT
                 }
+#ifdef TARGET_64BIT
                 else if ((tree->AsIntCon()->gtIconVal & 0xFFFFFFFF00000000LL) != 0)
                 {
-                    printf(" 0x%llx", dspIconVal);
-#endif
+                    if (dspIconVal >= 0)
+                    {
+                        printf(" 0x%llx", dspIconVal);
+                    }
+                    else
+                    {
+                        printf(" -0x%llx", -dspIconVal);
```

Comment:
Starting from VS2013 we have support for ["%zd"](https://docs.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=vs-2015#size-prefixes-for-printf-and-wprintf-format-type-specifiers) (and other runtime parts are already using it), so I would suggest to just replace this block with:
`printf(" %zd", dspIconVal);`
that will handle both 32/64 and negative/positive.

Note: gcc and clang have always supported that.

Reply:
I like see these numbers in hex, especially with my shared CSE constant changes where we often have to add or subtract a small offset.

Reply:
I have not seen hex numbers being printed with a negative sign, but maybe it is fine. Thanks for the explanation.